### PR TITLE
Add attr.Value access design doc.

### PR DESCRIPTION
**Pull request by paddycarver**
Add a design doc describing the problem of accessing a single
attribute's value, and proposing a solution for it.

### CODE REVIEW

**Review by bflad — APPROVED** [review]
This seems like the best path forward.

File: docs/design/attr-value-access.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+There are numerous places in the framework that require access to config,
+state, and plan values. In most places, like CRUD methods and the provider
+configuration callback, we provide access to the entire config, state, and
+plan. `Get` and `GetAttribute` methods on the config, state, and plan are then
+utilized to retrieve either the entire config/state/plan or to retrieve a
+specific attribute's value.
+
+There are some places in the framework, however, that are dealing with a
+specific attribute wtihin a plan, config, or state. Validation and plan
```
Comment (ewbankkit) [inline docs/design/attr-value-access.md:11]: Spelling.

```suggestion
specific attribute within a plan, config, or state. Validation and plan
```

File: docs/design/attr-value-access.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+reusability suffers. It's also uncomfortably similar to the
+`schema.ResourceData.Get` pattern that developers disliked so much. It's,
+overall, just a very brittle interface.
+
+It does, however, offer a slightly more unit testable option than the request
+helpers, as only the value needs to be in the request, not an entire resource.
+
+### Reflection helpers
+
+We can also add helpers, either to `tfsdk` or to `attr` to use the same
```
Comment (kmoe) [inline docs/design/attr-value-access.md:60]:
```suggestion
We can also add helpers, either to `tfsdk` or to `attr`, to use the same
```

**Review by kmoe — APPROVED** [review]
LGTM

**Pull request merged**